docs/docs/how-tos/fine-grained-permissions.md

```diff
@@ -0,0 +1,111 @@
+
+  ![Keycloak clients add jupyterhub role form](/img/how-tos/user_role_mapping_tab.png)
+
+- 3. Select the client for which role needs to be added to the user.
+
+  ![Keycloak clients add jupyterhub role form](/img/how-tos/user_role_mapping_roles.png)
+
+- 4. Select the role in the "Available Roles" and click on "Add Selected >>".
+
+  ![Keycloak clients add jupyterhub role form](/img/how-tos/user_role_mapping_add_role.png)
+
+Follow the same steps by clicking on the groups tab and selecting a group to attach a role to a
+group.
+
+In the above section, we learnt how to create a role with some attributes and attach that to a user or a group.
+Now we will learn how to create scopes to grant a particular set of permissions to the user.
+
+:::note
+After the roles are assigned to a user or group in keycloak, the user needs to login again to the service
+for the roles to take in effect. For example let's say we add a set of roles for conda-store to the user named
+"John Doe", now for the user "John Doe" to be able to avail newly granted/revoked roles, they need to login to
+conda-store again (similarly for jupyterhub as well), after the roles are granted/revoked.
+:::
+
+## Components Attribute
+
+We have seen in the above example the `component` attribute while creating a role. The value of this parameter
+depends on the type of component in the service, we're creating a role for, current we only have two components:
+
+- `jupyterhub`: to create jupyterhub native roles in `jupyterhub` client.
+- `conda-store`: to create conda-store roles in `conda_store` client
+
+## JupyterHub Scopes
+
+The syntax for `scopes` attribute for a jupyterhub role in Keycloak in Nebari follows the native RBAC scopes syntax
+for JupyterHub itself. The documentation for the same can be found [here](https://jupyterhub.readthedocs.io/en/stable/rbac/scopes.html#scope-conventions).
+
+Below is an example for the same.
+
+Scopes for allowing users to share apps in Nebari's jhub-apps launcher:
+
+`shares!user,read:users:name,read:groups:name`
+
+The `scopes` defined above consists of three scopes:
+
+- `shares!user`: grants permissions to share user's server
+- `read:users:name`: grants permissions to read other user's names
+- `read:groups:name`: grants permissions to read other groups's names
+
+To be able to share a server to a group or a user you need to be read other user's or group's names and must have
+permissions to be able to share your server, this is what this set of permissions implement.
+
+## Conda Store Scopes
+
+The scopes for roles for conda-store client describes level on permission on a namespace.
+
+Below are some examples for the same:
+
+`admin!namespace=analyst,developer!namespace=nebari-git`
+
+The `scopes` defined above consists of two scopes:
+
+- `admin!namespace=analyst`: grants `admin` access to namespace `analyst`
+- `developer!namespace=nebari-git`: grants `developer` access to namespace `nebari-git`
```

Comment:
Are `admin` and `developer` roles that exist within conda-store itself? Or are those `admin` and `developer` as we know it within Nebari's keycloak groups?

What does permissions do `admin` and `developer` have? In other words, read/write/delete access? 

And depending on how you'd answer those questions, more details around what that means within the namespace would be helpful as well. i.e. Can this user now read/create/delete environments in the namespace?

Reply:
>Are admin and developer roles that exist within conda-store itself? Or are those admin and developer as we know it within Nebari's keycloak groups?

Yes they do exists in conda-store: viewer, developer (or editor), admin

> What does permissions do admin and developer have? In other words, read/write/delete access?

What does permissions do admin and developer have? In other words, read/write/delete access?

`admin` has full access and `developer` (also same as `editor`) has edit access, here is the full list: https://github.com/conda-incubator/conda-store/blob/67a4860ac5d9cf33285b5641e5dbb3b9ce61b1c7/conda-store-server/conda_store_server/server/auth.py#L146

>And depending on how you'd answer those questions, more details around what that means within the namespace would be helpful as well. i.e. Can this user now read/create/delete environments in the namespace?

I'll add a few line summary and point to conda-store if that make sense.

Reply:
Yeah I think that sounds like a good solution